pkg/app2/listener.go

```diff
@@ -2,21 +2,22 @@ package app2
 
 import (
 	"net"
+	"sync"
 
 	"github.com/skycoin/skycoin/src/util/logging"
-
-	"github.com/skycoin/skywire/pkg/app2/network"
+	"github.com/skycoin/skywire/pkg/app2/appnet"
 )
 
 // Listener is a listener for app server connections.
 // Implements `net.Listener`.
 type Listener struct {
-	log     *logging.Logger
-	id      uint16
-	rpc     RPCClient
-	addr    network.Addr
-	cm      *idManager // contains conns associated with their IDs
-	freeLis func()
+	log       *logging.Logger
+	id        uint16
+	rpc       RPCClient
+	addr      appnet.Addr
+	cm        *idManager // contains conns associated with their IDs
+	freeLis   func()
+	freeLisMx sync.RWMutex
```

Comment:
@evanlinjin this is true, but we have a possible data race. `free` is being returned when we add `conn` to the manager. In usual circumstances, client code doesn't have access to the `Conn` or `Listener` until the function creating it returns, so there's no possibility for the data race. But we're dealing with the client code here, it may pass strange ids to the methods. If client code calls `CloseConn` with the id of a conn which is added to the manager, but not yet filled with the `free` func (`Accept` didn't return yet, for example), we will have a data race. If you insist, I may remove this, but I strongly recommend keeping it. Of course panic occurs here because of the data race, it will be a client's fault, but this mutex make things more robust

Reply:
@Darkren in this case, it may be a good idea to put in a comment here. I have yet to explore your chain of thought in regards to this possible data race. Let's make a comment and we will give it further discussion later on.

Reply:
Closed all similair queries, left this one for future discussion